Source/ToolCore/Build/AndroidProjectGenerator.cpp

```diff
@@ -281,6 +313,55 @@ bool AndroidProjectGenerator::GenerateAndroidManifest()
 
 }
 
+bool AndroidProjectGenerator::CopyUserIcons()
+{
+    FileSystem* fileSystem = GetSubsystem<FileSystem>();
+    ToolSystem* toolSystem = GetSubsystem<ToolSystem>();
+    Project* project = toolSystem->GetProject();
+    AndroidBuildSettings* settings = project->GetBuildSettings()->GetAndroidBuildSettings();
+
+    String userIconPath = settings->GetIconPath();
+    if (!fileSystem->DirExists(userIconPath))               // dont do anything if there is no path defined.
+        return true;
+            
+    String userIconDir = userIconPath + "/drawable";        // 1st target dir 
+    String userIconFile = userIconDir + "/logo_large.png";  // 1st target file
+    String destDir = buildPath_ + "/res/drawable";          // where it should be in the build
+    if ( fileSystem->FileExists (userIconFile) )            // is there a file there?
+    {
+        Poco::File icon1 ( userIconFile.CString() );        // declare the file
+        icon1.copyTo( destDir.CString() );                  // copy the file into the build
```

Comment:
Awesome, my C++ visibility knowledge is failing this morning, if we made the Android project generator a friend of the Android build class would it have access to the protected BuildBase methods?  If so, maybe opt for this... and sounds great on the audit.